### PR TITLE
Kafka 0.10 support

### PR DESCRIPTION
**Pull request by smferguson**
expecting (hoping) to start a conversation with this. my company's going to want (need?) 0.10 support in the very near future so I'd like to hear thoughts/musings around how it might be accomplished.

### CODE REVIEW

**Comment by osheroff** [comment]
I think this is fine.  I can merge as is after some minimal testing (provided that the thing compiles with 0.10.0).  I think @dadah89 put some code in .travis.yml to at least ensure the code compiles against each version of the kafka-clients library, you may want to make sure that that is still working and is compiling maxwell against 0.10.0.

Maybe in a world with more time Maxwell could probe the kafka cluster, discover the version and then automatically choose the right version of the producer library.  But a read-through of the protocol doesn't really seem to indicate a reliable way of determining the cluster version (at least, not straightforward), so this way will do just fine for now.

**Comment by osheroff** [comment]
ok, this *looks* good, but falls on its face pretty hard:

```
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/kafka/common/serialization/Serializer
```

meaning the 0.9 and 0.10 libraries aren't interface compatible.  So we'll either have to come up with a solution that works with both interfaces, or we'll have to compile a little Maxwell->Kafka adapter class for each version of kafka.  

the adapter may be the right solution long term anyway, not sure.

**Comment by osheroff** [comment]
never mind that last comment.  I just didn't compile the damn thing correctly.

**Comment by osheroff** [comment]
I'm back to "this looks fine", although:

- the options need documenting (and should probably just remove the --kafka0.8 option
- it should support both `--kafka_version 0.9` and `--kafka_version=0.9`

**Comment by smferguson** [comment]
related to issue #363

File: docs/docs/config.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ kafka_partition_columns        | STRING                              | if partit
 kafka_partition_by_fallback    | [database┃table┃primary_key]        | required when kafka_partition_by=column.  Used when the column is missing |
 kafka_partition_hash           | [default┃murmur3]                   | hash function to use when hoosing kafka partition   | default
 ddl_kafka_topic                | STRING                              | if output_ddl is true, kafka topic to write DDL changes to | *kafka_topic*
-kafka0.8                       | BOOLEAN                             | run maxwell with kafka producer 0.8 (instead of the 0.9) |
+kafka_version                  | [0.8|0.9|0.10]                      | run maxwell with kafka producer 0.8.2, 0.9.0 or 0.10.0 | 0.9.0
```
Comment (osheroff) [inline docs/docs/config.md]: careful, the pipes here will break the table formatting (sadly).  try copy/pasting the funky looking pipe character from above.  Or if you have a better solution I'm all ears.

Comment (osheroff) [inline docs/docs/config.md]: also, check your doc changes with `mkdocs serve`, after installing the python 'mkdocs' package

Comment (smferguson) [inline docs/docs/config.md]: 👍
internets say use "&#124;". i'm messing with that at the moment.

**Comment by smferguson** [comment]
lmk what you think of the alternative ```&#124;```

**Comment by smferguson** [comment]
@osheroff hang on for a bit. think i found a bug.

**Comment by smferguson** [comment]
@osheroff ok. feeling better about this now.

**Comment by osheroff** [comment]
ok.  There's bonus points here for exposing the kafka version in `MaxwellConfig` or something so we can make https://github.com/zendesk/maxwell/issues/451 go away.  But this looks totally fine to me.  👍 and will merge after test

**Comment by smferguson** [comment]
feature creeeeeep! seriously tho. I'll try to take a look at it this week, but can't promise anything. adding 0.10 (along with the other pieces) gets us to the point we can start profiling maxwell/kafka to understand its performance/monitoring needs before we push to prod.
you guys should write a blog about the scale you run maxwell at/what it's like living with over time ;)

**Pull request merged**